Source/Evolve/WarpXEvolve.cpp

```diff
@@ -303,6 +302,18 @@ WarpX::OneStep_nosub (Real cur_time)
     if (warpx_py_particlescraper) warpx_py_particlescraper();
     if (warpx_py_beforedeposition) warpx_py_beforedeposition();
     PushParticlesandDepose(cur_time);
+
+    // If warpx.do_current_centering = 1, center currents from nodal grid to staggered grid
+    if (WarpX::do_current_centering)
+    {
+        for (int lev = 0; lev <= finest_level; lev++)
+        {
+            WarpX::UpdateCurrentNodalToStag(*current_fp[lev][0], *current_fp_nodal[lev][0]);
+            WarpX::UpdateCurrentNodalToStag(*current_fp[lev][1], *current_fp_nodal[lev][1]);
+            WarpX::UpdateCurrentNodalToStag(*current_fp[lev][2], *current_fp_nodal[lev][2]);
+        }
+    }
```

Comment:
Could this be added inside the function `SyncCurrent`?
This is because SyncCurrent is performs other, somewhat related operations (e.g. filtering).

Reply:
Absolutely, done in https://github.com/ECP-WarpX/WarpX/pull/1763/commits/c822fb1eea49d62050934257b6e86fe8b6aa69a5.